src/common/common.c

```diff
@@ -71,6 +71,27 @@ static void set_available_cpu_extensions(void) {
 	cpu_ext_data[OQS_CPU_EXT_INIT] = 1;
 }
 #elif defined(OQS_DIST_ARM64v8_BUILD)
+#if defined(__APPLE__)
+#include <sys/sysctl.h>
+static int macos_feature_detection(const char *feature_name) {
+	int p;
+	size_t p_len = sizeof(p);
+	int res = sysctlbyname(feature_name, &p, &p_len, NULL, 0);
+	if (res != 0) {
+		return 0;
+	} else {
+		return (p != 0) ? 1 : 0;
+	}
+}
+static void set_available_cpu_extensions(void) {
+	/* mark that this function has been called */
+	cpu_ext_data[OQS_CPU_EXT_ARM_AES] = 1;
+	cpu_ext_data[OQS_CPU_EXT_ARM_SHA2] = 1;
+	cpu_ext_data[OQS_CPU_EXT_ARM_SHA3] = macos_feature_detection("hw.optional.armv8_2_sha3");
+	cpu_ext_data[OQS_CPU_EXT_ARM_NEON] = macos_feature_detection("hw.optional.neon");
```

Comment:
Should `macos_feature_detection` return an unisnged int? Currently, this causes warnings for implicit conversion

Reply:
Looking at the function, I think returning an unsigned int makes the most sense. I've pushed an updated version and am testing it on all the arm platforms I have access to.